index.js.flow

```diff
@@ -0,0 +1,175 @@
+/* @flow */
+
+/**
+ * Misc Setup Types
+ */
+
+declare type PromiseLike<R> = {
+  then<U>(
+    onFulfill?: (value: R) => Promise<U> | U,
+    onReject?: (error: any) => Promise<U> | U
+  ): Promise<U>;
+}
```

Comment:
Is there any plan to have this builtin in Flow? Kinda weird having to define it manually in every type definition.